src/Protocol/Protocol_1_13.cpp

```diff
@@ -460,5 +792,635 @@ void cProtocol_1_13::WriteItem(cPacketizer & a_Pkt, const cItem & a_Item)
+					break;
+				}
+				case cProjectileEntity::pkFirework:
+				{
+					// TODO
+					break;
+				}
+				case cProjectileEntity::pkSplashPotion:
+				{
+					// TODO
+				}
+				default:
+				{
+					break;
+				}
+			}
+			break;
+		}  // case etProjectile
+
+		case cEntity::etMonster:
+		{
+			WriteMobMetadata(a_Pkt, static_cast<const cMonster &>(a_Entity));
+			break;
+		}
+
+		case cEntity::etBoat:
+		{
+			auto & Boat = static_cast<const cBoat &>(a_Entity);
+
+			a_Pkt.WriteBEInt8(BOAT_LAST_HIT_TIME);
+			a_Pkt.WriteBEInt8(METADATA_TYPE_VARINT);
+			a_Pkt.WriteVarInt32(static_cast<UInt32>(Boat.GetLastDamage()));
+
+			a_Pkt.WriteBEInt8(BOAT_FORWARD_DIRECTION);
+			a_Pkt.WriteBEInt8(METADATA_TYPE_VARINT);
+			a_Pkt.WriteVarInt32(static_cast<UInt32>(Boat.GetForwardDirection()));
+
+			a_Pkt.WriteBEInt8(BOAT_DAMAGE_TAKEN);
+			a_Pkt.WriteBEInt8(METADATA_TYPE_FLOAT);
+			a_Pkt.WriteBEFloat(Boat.GetDamageTaken());
+
+			a_Pkt.WriteBEInt8(BOAT_TYPE);
+			a_Pkt.WriteBEInt8(METADATA_TYPE_VARINT);
+			a_Pkt.WriteVarInt32(static_cast<UInt32>(Boat.GetMaterial()));
+
+			// TODO: some issue with these, investigate
+			/*
+			a_Pkt.WriteBEInt8(BOAT_RIGHT_PADDLE_TURNING);
+			a_Pkt.WriteBEInt8(METADATA_TYPE_BOOL);
+			a_Pkt.WriteBool(Boat.IsRightPaddleUsed());
+
+			a_Pkt.WriteBEInt8(BOAT_LEFT_PADDLE_TURNING);
+			a_Pkt.WriteBEInt8(METADATA_TYPE_BOOL);
+			a_Pkt.WriteBool(static_cast<bool>(Boat.IsLeftPaddleUsed()));
+
+			// TODO: splash timer?
+			a_Pkt.WriteBEInt8(BOAT_SPLASH_TIMER);
+			a_Pkt.WriteBEInt8(METADATA_TYPE_VARINT);
+			a_Pkt.WriteVarInt32(0);
+			*/
+
+			break;
+		}  // case etBoat
+
+		case cEntity::etItemFrame:
+		{
+			// TODO
+			break;
+		}  // case etItemFrame
+
+		default:
+		{
+			break;
+		}
+	}
+}
+
+
+
+
+
+void cProtocol_1_13::WriteMobMetadata(cPacketizer & a_Pkt, const cMonster & a_Mob)
+{
+	using namespace Metadata;
+
+	// Living Enitiy Metadata
+	if (a_Mob.HasCustomName())
+	{
+		// TODO: As of 1.9 _all_ entities can have custom names; should this be moved up?
+		a_Pkt.WriteBEUInt8(ENTITY_CUSTOM_NAME);
+		a_Pkt.WriteBEUInt8(METADATA_TYPE_STRING);
+		a_Pkt.WriteString(a_Mob.GetCustomName());
+
+		a_Pkt.WriteBEUInt8(ENTITY_CUSTOM_NAME_VISIBLE);  // Custom name always visible
+		a_Pkt.WriteBEUInt8(METADATA_TYPE_BOOL);
+		a_Pkt.WriteBool(a_Mob.IsCustomNameAlwaysVisible());
+	}
+
+	a_Pkt.WriteBEUInt8(LIVING_HEALTH);
+	a_Pkt.WriteBEUInt8(METADATA_TYPE_FLOAT);
+	a_Pkt.WriteBEFloat(static_cast<float>(a_Mob.GetHealth()));
+
+	switch (a_Mob.GetMobType())
+	{
+		case mtBat:
+		{
+			auto & Bat = static_cast<const cBat &>(a_Mob);
+			a_Pkt.WriteBEUInt8(BAT_HANGING);
+			a_Pkt.WriteBEUInt8(METADATA_TYPE_BYTE);
+			a_Pkt.WriteBEInt8(Bat.IsHanging() ? 1 : 0);
+			break;
+		}  // case mtBat
+
+		case mtChicken:
+		{
+			/* "ata cannot be cast to java.lang.Boolean"? Investigate
+			auto & Chicken = static_cast<const cChicken &>(a_Mob);
+
+			a_Pkt.WriteBEUInt8(AGEABLE_BABY);
+			a_Pkt.WriteBEUInt8(METADATA_TYPE_BOOL);
```

Comment:
They added a new metadata type so `METADATA_TYPE_BOOL` needs updating, plus this field should now be a `WriteVarInt32`. Ideally, if metadata largely hasn't changed, we need to figure out a way to not copy paste the entire function XD

https://wiki.vg/index.php?title=Entity_metadata&direction=next&oldid=14048#Entity_Metadata_Format

Reply:
Thanks! This probably explains some other weird behavior I was experiencing.

Reply:
I don't remember seeing any large metadata changes when I last worked on this PR. The only real problem was the boolean one, but that can be fixed easily now that we know the cause.